zookeeper-api/src/main/java/org/apache/helix/zookeeper/zkclient/ZkClient.java

```diff
@@ -240,17 +240,18 @@ protected ZkClient(IZkConnection zkConnection, int connectionTimeout, long opera
       LOG.info("ZkClient monitor key or type is not provided. Skip monitoring.");
     }
 
-    connect(connectionTimeout, this);
-
-    try {
```

Comment:
why monitoring related code removed?

Reply:
This is moved to inside `connect()`, you can look at the code below. Those two things should be combined and always happen together.